### DISCUSSION TITLE
Not getting CPU clock on Windows 11

### DISCUSSION THREAD

**Discussion by mp009** [LibreHardwareMonitor/LibreHardwareMonitor#1457] [General]
I'm not sure if is only me (i've tested on 3 different machines) but not getting CPU clock on Windows 11 (version 10.0.22631.0)

**Comment by mp009** [comment]
for same conf file, On a different version of windows I'm getting: 
```
Load CPU Core #1 Thread #1
Load CPU Core #1 Thread #2
Load CPU Core #2 Thread #1
Load CPU Core #2 Thread #2
Load CPU Core #3 Thread #1
Load CPU Core #3 Thread #2
Load CPU Core #4 Thread #1
Load CPU Core #4 Thread #2
Load CPU Core #5 Thread #1
Load CPU Core #5 Thread #2
Load CPU Core #6 Thread #1
Load CPU Core #6 Thread #2
Load CPU Core #7
Load CPU Core #8
Load CPU Core #9
Load CPU Core #10
Load CPU Core #11
Load CPU Core #12
Load CPU Core #13
Load CPU Core #14
Load CPU Total
Load CPU Core Max
Temperature CPU Core #1
Temperature CPU Core #2
Temperature CPU Core #3
Temperature CPU Core #4
Temperature CPU Core #5
Temperature CPU Core #6
Temperature CPU Core #7
Temperature CPU Core #8
Temperature CPU Core #9
Temperature CPU Core #10
Temperature CPU Core #11
Temperature CPU Core #12
Temperature CPU Core #13
Temperature CPU Core #14
Temperature CPU Package
Temperature CPU Core #1 Distance to TjMax
Temperature CPU Core #2 Distance to TjMax
Temperature CPU Core #3 Distance to TjMax
Temperature CPU Core #4 Distance to TjMax
Temperature CPU Core #5 Distance to TjMax
Temperature CPU Core #6 Distance to TjMax
Temperature CPU Core #7 Distance to TjMax
Temperature CPU Core #8 Distance to TjMax
Temperature CPU Core #9 Distance to TjMax
Temperature CPU Core #10 Distance to TjMax
Temperature CPU Core #11 Distance to TjMax
Temperature CPU Core #12 Distance to TjMax
Temperature CPU Core #13 Distance to TjMax
Temperature CPU Core #14 Distance to TjMax
Temperature Core Max
Temperature Core Average
Clock CPU Core #1
Clock CPU Core #2
Clock CPU Core #3
Clock CPU Core #4
Clock CPU Core #5
Clock CPU Core #6
Clock CPU Core #7
Clock CPU Core #8
Clock CPU Core #9
Clock CPU Core #10
Clock CPU Core #11
Clock CPU Core #12
Clock CPU Core #13
Clock CPU Core #14
Power CPU Package
Power CPU Cores
Power CPU Memory
Voltage CPU Core
Voltage CPU Core #1
Voltage CPU Core #2
Voltage CPU Core #3
Voltage CPU Core #4
Voltage CPU Core #5
Voltage CPU Core #6
Voltage CPU Core #7
Voltage CPU Core #8
Voltage CPU Core #9
Voltage CPU Core #10
Voltage CPU Core #11
Voltage CPU Core #12
Voltage CPU Core #13
Voltage CPU Core #14
Clock Bus Speed
``` 

But on Windows 11 I'm getting these:
```
Load CPU Core #1
Load CPU Core #2
Load CPU Core #3
Load CPU Core #4
Load CPU Core #5
Load CPU Core #6
Load CPU Core #7
Load CPU Core #8
Load CPU Core #9 Thread #1
Load CPU Core #9 Thread #2
Load CPU Core #10 Thread #1
Load CPU Core #10 Thread #2
Load CPU Core #11
Load CPU Core #12
Load CPU Total
Load CPU Core Max
Voltage CPU Core
Voltage CPU Core #1
Voltage CPU Core #2
Voltage CPU Core #3
Voltage CPU Core #4
Voltage CPU Core #5
Voltage CPU Core #6
Voltage CPU Core #7
Voltage CPU Core #8
Voltage CPU Core #9
Voltage CPU Core #10
Temperature Temperature
```

**Comment by Xcalibur37** [comment]
In this case, it looks like you are using different processors. What processors are you using for each screen respectively? Could be a limitation on what Libre can return based on your Windows 11 processor version.

**Comment by mp009** [comment]
On Windows 11 I have Intel(R) Core(TM) Ultra 7 165U   1.70 GHz

**Reply by Xcalibur37** [reply]
Considering that should register as a Meteor Lake it should come up similarly as any other 14th gen processor. Without having direct access to this processor, are you sure you are running elevated? This includes making sure the package is unblocked at the ZIP level before unpacking to make sure Windows doesn't lock anything down (which is super annoying).

**Reply by mp009** [reply]
thanks for looking into this, yes, I'm running in elevated, in fact the latest prev release looks like it fixes :

`
CPU Core #1 - Clock - 1493.533
CPU Core #2 - Clock - 2489.221
CPU Core #3 - Clock - 2489.221
CPU Core #4 - Clock - 1393.964
CPU Core #5 - Clock - 1493.533
CPU Core #6 - Clock - 1393.964
CPU Core #7 - Clock - 1393.964
CPU Core #8 - Clock - 1393.964
CPU Core #9 - Clock - 4281.46
CPU Core #10 - Clock - 4878.873
CPU Core #11 - Clock - 1095.257
CPU Core #12 - Clock - 1095.257
`

**Comment by Xcalibur37** [comment]
Digging into the diff between versions - looks like the previous version didn't have Meteor Lake or the first code for Arrow Lake (Gen. 15) - which makes sense given the timings of the releases.  So, in this case it would use the "default" value of the switch statement which is different than what the known processors would call.  Mind you I am not one of the authors of this code so I am digging into this to see if I can figure it out.

I guess what I would like to know is - does your processor actually register as "Meteor Lake"? Researching the spec from Intel it's labeled as "Formally Meteor Lake" so it should have the same signature and in turn your Core Ultra should behave similarly to the standard Meteor Lake family. Clearly this isn't true which is baffling.

**To the broader community;**
Is there anyone else on the forums here with a Core Ultra that can help us test this finding? Could be an interesting nuance of the Core Ultra family that may need a work around in the next version.

In the meantime, how comfortable are you stepping through the code in Visual Studio? Is that something you could do or are you more comfortable as just a power user. I would like to see how this steps through the **_LibreHardwareMonitorLib/Hardware/Cpu/IntelCpu.cs_**.

**Comment by mejustandrew** [comment]
I've had this issue on an 165 CPU. Debugging it, there was an error code 5 returned by OpenSCManager method call on advapi32.dll. Digging further, this came from lack of permissions. Running as Admin from a location with Admin rights allowed me to get the temperature data.
